src/tests/ftest/soak/soak.py

```diff
@@ -65,120 +72,228 @@ def job_done(self, args):
-                remote_dir (str): remote directory of files
-                local_dir (str): local directory
-
         Raises:
             SoakTestError: if there is an error with the remote copy
 
         """
+        # copy the files from the remote
         this_host = socket.gethostname()
-        result = pcmd(
-            NodeSet.fromlist(hostlist),
-            "if [ ! -z '$(ls -A {0})' ]; then "
-            "scp -p -r {0}/ \"{1}:'{2}/'\" && rm -rf {0}/*; fi".format(
-                remote_dir, this_host, local_dir),
-            verbose=False)
-        if len(result) > 1 or 0 not in result:
+        result = slurm_utils.srun(
+            self.partition_clients,
+            NodeSet.fromlist(self.hostlist_clients),
+            "bash -c \"scp -p -r {0} {1}:{0}/.. && rm -rf {0}/*\"".format(
+                self.test_log_dir, this_host))
+        if result == 0:
+            cmd = "cp -R {0}/ \'{1}\'; rm -rf {0}/*".format(
+                self.test_log_dir, self.outputsoakdir)
+            try:
+                result = process.run(cmd, shell=True, timeout=30)
+            except process.CmdError as error:
+                raise SoakTestError(
+                    "<<FAILED: Soak remote logfiles not copied"
+                    "to avocado data dir {} - check /tmp/soak "
+                    "on nodes {}>>".format(error, self.hostlist_clients))
+        else:
             raise SoakTestError(
-                "Error executing remote copy: {}".format(
-                    ", ".join(
-                        [str(result[key]) for key in result if key != 0])))
+                "<<FAILED: Soak remote logfiles not copied "
+                "from clients>>: {}".format(self.hostlist_clients))
+
+    def is_harasser(self, harasser):
+        """Check if harasser is defined in yaml.
+
+        Args:
+            harasser (list): list of harassers to launch
+
+        Returns: bool
+
+        """
+        return self.h_list and harasser in self.h_list
+
+    def launch_harassers(self, harassers, pools):
+        """Launch any harasser tests if defined in yaml.
 
-    def create_ior_cmdline(self, job_params, job_spec, pool):
+        Args:
+            harasser (list): list of harassers to launch
+            pools (TestPool): pool obj
+
+        """
+        job = []
+        # Launch harasser after one complete pass
+        for harasser in harassers:
+            if "rebuild" == harasser:
+                method = self.launch_rebuild
+                ranks = self.params.get(
+                    "ranks_to_kill", "/run/" + harasser + "/*")
+                targets = self.params.get("targets", "/run/server_config/*")
+                param_list = (ranks, pools, targets)
+                name = "REBUILD"
+            else:
+                raise SoakTestError(
+                    "<<FAILED: Harasser {} is not supported. ".format(
+                        harasser))
+            job = threading.Thread(
```

Comment:
(pylint-redefined-variable-type) Redefinition of job type from list to threading.Thread